File: internal/plm/entity/cmf.go
package entity

import "time"

// CMFDesign 一个外观零件的一种CMF方案
type CMFDesign struct {
	ID                  string    `json:"id" gorm:"primaryKey;size:32"`
	ProjectID           string    `json:"project_id" gorm:"size:32;not null;index"`
	TaskID              string    `json:"task_id" gorm:"size:32;not null;index"`
	BOMItemID           string    `json:"bom_item_id" gorm:"size:32;not null;index"`
	SchemeName          string    `json:"scheme_name" gorm:"size:64"`
	Color               string    `json:"color" gorm:"size:64"`
	ColorCode           string    `json:"color_code" gorm:"size:64"`
	GlossLevel          string    `json:"gloss_level" gorm:"size:32"`  // 高光/半哑/哑光/丝光/镜面
	SurfaceTreatment    string    `json:"surface_treatment" gorm:"size:128"`
	TexturePattern      string    `json:"texture_pattern" gorm:"size:64"` // 皮纹/磨砂/拉丝
	CoatingType         string    `json:"coating_type" gorm:"size:64"`    // UV漆/PU漆/粉末涂装
	RenderImageFileID   *string   `json:"render_image_file_id" gorm:"size:32"`
	RenderImageFileName string    `json:"render_image_file_name" gorm:"size:256"`
	Notes               string    `json:"notes" gorm:"type:text"`
	SortOrder           int       `json:"sort_order" gorm:"default:0"`
	CreatedAt           time.Time `json:"created_at"`
	UpdatedAt           time.Time `json:"updated_at"`

	BOMItem  *ProjectBOMItem `json:"bom_item,omitempty" gorm:"foreignKey:BOMItemID"`
	Drawings []CMFDrawing    `json:"drawings,omitempty" gorm:"foreignKey:CMFDesignID"`
}

// CMFDrawing CMF方案的工艺图纸
type CMFDrawing struct {
	ID          string    `json:"id" gorm:"primaryKey;size:32"`
	CMFDesignID string    `json:"cmf_design_id" gorm:"size:32;not null;index"`
	DrawingType string    `json:"drawing_type" gorm:"size:32"` // 丝印/激光雷雕/UV转印/移印/烫金
	FileID      string    `json:"file_id" gorm:"size:32;not null"`
	FileName    string    `json:"file_name" gorm:"size:256;not null"`
	Notes       string    `json:"notes" gorm:"type:text"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: nimo-plm-web/src/api/cmf.ts
import apiClient from './client';

export interface CMFDesign {
  id: string;
  project_id: string;
  task_id: string;
  bom_item_id: string;
  scheme_name: string;
  color: string;
  color_code: string;
  gloss_level: string;
  surface_treatment: string;
  texture_pattern: string;
  coating_type: string;
  render_image_file_id?: string;
  render_image_file_name?: string;
  notes: string;
  sort_order: number;
  created_at: string;
  updated_at: string;
  bom_item?: {
    id: string;
    name: string;
    item_number: number;
    material_type?: string;
    thumbnail_url?: string;
  };
  drawings?: CMFDrawing[];
}

export interface CMFDrawing {
  id: string;
  cmf_design_id: string;
  drawing_type: string;
  file_id: string;
  file_name: string;
  notes: string;
  created_at: string;
  updated_at: string;
}

export interface AppearancePart {
  id: string;
  bom_id: string;
  item_number: number;
  name: string;
  material_type?: string;
  thumbnail_url?: string;
  process_type?: string;
}

export const cmfApi = {
  // 获取外观件列表
  getAppearanceParts: async (projectId: string, taskId: string): Promise<AppearancePart[]> => {
    const res = await apiClient.get(`/projects/${projectId}/tasks/${taskId}/cmf/appearance-parts`);
    return res.data.data?.items || [];
  },

  // 列出任务的所有CMF方案
  listDesigns: async (projectId: string, taskId: string): Promise<CMFDesign[]> => {
    const res = await apiClient.get(`/projects/${projectId}/tasks/${taskId}/cmf/designs`);
    return res.data.data?.items || [];
  },

  // 列出项目所有CMF方案
  listDesignsByProject: async (projectId: string): Promise<CMFDesign[]> => {
    const res = await apiClient.get(`/projects/${projectId}/cmf/designs`);
    return res.data.data?.items || [];
  },

  // 创建CMF方案
  createDesign: async (projectId: string, taskId: string, data: Partial<CMFDesign>): Promise<CMFDesign> => {
    const res = await apiClient.post(`/projects/${projectId}/tasks/${taskId}/cmf/designs`, data);
    return res.data.data;
  },

  // 更新CMF方案
  updateDesign: async (projectId: string, taskId: string, designId: string, data: Partial<CMFDesign>): Promise<CMFDesign> => {
    const res = await apiClient.put(`/projects/${projectId}/tasks/${taskId}/cmf/designs/${designId}`, data);
    return res.data.data;
  },

  // 删除CMF方案
  deleteDesign: async (projectId: string, taskId: string, designId: string): Promise<void> => {
    await apiClient.delete(`/projects/${projectId}/tasks/${taskId}/cmf/designs/${designId}`);
  },

  // 添加图纸
  addDrawing: async (designId: string, data: { drawing_type: string; file_id: string; file_name: string; notes?: string }): Promise<CMFDrawing> => {
    const res = await apiClient.post(`/cmf-designs/${designId}/drawings`, data);
    return res.data.data;
  },

  // 删除图纸
  removeDrawing: async (designId: string, drawingId: string): Promise<void> => {
    await apiClient.delete(`/cmf-designs/${designId}/drawings/${drawingId}`);
  },
};
